onnxruntime/contrib_ops/cpu/quantization/dynamic_quantize_matmul.cc

```diff
@@ -161,6 +162,122 @@ class DynamicQuantizeMatMul final : public MatMulIntegerToFloatBase {
 
   Status Compute(OpKernelContext* context) const override;
 
+#if defined(USE_KLEIDIAI) && !defined(_MSC_VER)
+  Status PrePack(const Tensor& tensor, int input_idx, AllocatorPtr alloc,
+                 /*out*/ bool& is_packed,
+                 /*out*/ PrePackedWeights* prepacked_weights) override {
+    // only pack Matrix B
+    if (input_idx == GetBIdx()) {
+      const Tensor* b_zp_constant_tensor{nullptr};
+      bool b_quantization_is_asymmetric = false;
+
+      // zero point tensor could be provided as a direct input to the kernel and not as a constant so this
+      // test is not sufficient
+      const OrtValue* b_zp;
+      if (Info().TryGetConstantInput(IN_B_ZERO_POINT, &b_zp)) {
+        b_zp_constant_tensor = &b_zp->Get<Tensor>();
+      }
+
+      // MlasDynamicQgemm requires symmetric quantization for B, so no zero point should exist or it should
+      // have a zero value
+      if (b_zp_constant_tensor != nullptr) {  // Covers the case where tensor is not a constant
```

Comment:
I still suggest handling the case where the B zp tensor is not constant but not provided.

```c++
if (b_zp_constant_tensor != nullptr) {
    // ...
} else {
    // B zero point input is not constant. If it exists, we can't assume symmetric quantization.
    const auto input_defs = Info().node().InputDefs();
    const bool b_zp_input_exists = input_defs.size() > IN_B_ZERO_POINT && input_defs[IN_B_ZERO_POINT]->Exists();
    b_quantization_is_asymmetric = b_zp_input_exists;
}
```